pkg/svc/zserver.go

```diff
@@ -2,51 +2,206 @@ package svc
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.GetDescription(ctx, &rpb.GetDescriptionRequest{Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetDescriptionResponse{Description: resp.Description}, nil
+
+	}
+	s.IncRequests()
+	return &pb.GetDescriptionResponse{Description: s.GetDescriptionFromServer()}, nil
+}
+
+// Returns uptime in seconds
+func (s *server) GetUptime(ctx context.Context, req *pb.GetUptimeRequest) (*pb.GetUptimeResponse, error) {
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.GetUptime(ctx, &rpb.GetUptimeRequest{Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetUptimeResponse{Uptime: resp.Uptime}, nil
+
+	}
+	s.IncRequests()
+	uptime := time.Now().Unix() - s.Timestamp.Unix()
+	return &pb.GetUptimeResponse{Uptime: uptime}, nil
+}
+
+func (s *server) GetRequests(ctx context.Context, req *pb.GetRequestsRequest) (*pb.GetRequestsResponse, error) {
+	if req.GetService() != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		resp, err := client.GetRequests(ctx, &rpb.GetRequestsRequest{Service: req.Service})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.GetRequestsResponse{Requests: resp.Requests}, nil
+
+	}
+	s.IncRequests()
+	return &pb.GetRequestsResponse{Requests: int64(s.GetRequestsFromServer())}, nil
+}
+
+func (s *server) GetMode(ctx context.Context, req *pb.GetModeRequest) (*pb.GetModeResponse, error) {
+	conn, err := GRPCConnect()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	client := rpb.NewMyResponderClient(conn)
+	resp, err := client.GetMode(ctx, &rpb.GetModeRequest{})
+	if err != nil {
+		return nil, err
+	}
+	s.IncRequests()
+	return &pb.GetModeResponse{Mode: resp.Mode}, nil
+}
+
+func (s *server) SetMode(ctx context.Context, req *pb.SetModeRequest) (*pb.SetModeResponse, error) {
+	conn, err := GRPCConnect()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	client := rpb.NewMyResponderClient(conn)
+	resp, err := client.SetMode(ctx, &rpb.SetModeRequest{Mode: req.Mode})
+	if err != nil {
+		return nil, err
+	}
+	s.IncRequests()
+	return &pb.SetModeResponse{Mode: resp.Mode}, nil
+}
+
+func (s *server) Restart(ctx context.Context, req *pb.RestartRequest) (*pb.RestartResponse, error) {
+	if req.Service != 1 {
+		conn, err := GRPCConnect()
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+		client := rpb.NewMyResponderClient(conn)
+		_, err = client.Restart(ctx, &rpb.RestartRequest{})
+		if err != nil {
+			return nil, err
+		}
+		return &pb.RestartResponse{}, nil
+	}
+	s.Description = viper.GetString("app.id")
```

Comment:
Concurrency

Reply:
it is just a mock function, in real cluster it should just terminate service, so i thought, that its not required to think about concurrency here

Reply:
I would say even if it is a mock function from high level design point, since it's a part of your service's logic it should be compatible with other parts of it.